### PR TITLE
{data}[foss/2019a] MariaDB-10.3.14 and dependencies

### PR DESCRIPTION
**Pull request by lexming**
Update for MariaDB to version 10.3.14. This build enables most compressors and engines, including RocksDB, TokuDB, Connect XML and OQGraph. Dependencies specific for each engine are indicated. Jemalloc is enabled by default, which after some quick benchmarks offers better performance than GNU malloc. We have been running this build for more than a month with no issues.

### CODE REVIEW

**Comment by migueldiascosta** [comment]
@lexming please remove the `Boost-1.70.0-foss-2019a.eb.bak_20190707184848_68757` file

File: easybuild/easyconfigs/b/Boost/Boost-1.70.0-foss-2019a.eb [hunk 1]
```diff
@@ -0,0 +1,24 @@
+name = 'Boost'
+version = '1.70.0'
+
+homepage = 'http://www.boost.org/'
+description = "Boost provides free peer-reviewed portable C++ source libraries."
+
+toolchain = {'name': 'foss', 'version': '2019a'}
```
Comment (migueldiascosta) [inline easybuild/easyconfigs/b/Boost/Boost-1.70.0-foss-2019a.eb]: why not use the existing easyconfig for `gompi`, which is a subtoolchain of `foss`?

File: easybuild/easyconfigs/m/MariaDB/MariaDB-10.3.14-foss-2019a.eb [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+name = 'MariaDB'
+version = '10.3.14'
+
+homepage = 'https://mariadb.org/'
+description = """MariaDB is an enhanced, drop-in replacement for MySQL.
+Included engines: myISAM, Aria, InnoDB, RocksDB, TokuDB, OQGraph, Mroonga."""
+
+toolchain = {'name': 'foss', 'version': '2019a'}
+
+source_urls = ['http://ftp.hosteurope.de/mirror/archive.mariadb.org/mariadb-%(version)s/source']
```
Comment (migueldiascosta) [inline easybuild/easyconfigs/m/MariaDB/MariaDB-10.3.14-foss-2019a.eb]: why use a mirror?

File: easybuild/easyconfigs/m/MariaDB/MariaDB-10.3.14-foss-2019a.eb [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+sources = [SOURCELOWER_TAR_GZ]
```
Comment (migueldiascosta) [inline easybuild/easyconfigs/m/MariaDB/MariaDB-10.3.14-foss-2019a.eb]: are the submodules (in the `10.3.10` easyconfig) no longer necessary?

File: easybuild/easyconfigs/b/Boost/Boost-1.70.0-foss-2019a.eb (hunk 1 above)
Comment (lexming) [reply easybuild/easyconfigs/b/Boost/Boost-1.70.0-foss-2019a.eb]: You're right, that's absolutely not necessary. I didn't know that `gompi` was a subtoolchain of `foss`.

File: easybuild/easyconfigs/m/MariaDB/MariaDB-10.3.14-foss-2019a.eb (hunk 2 above)
Comment (lexming) [reply easybuild/easyconfigs/m/MariaDB/MariaDB-10.3.14-foss-2019a.eb]: There is no public hosting of sources at mariadb.org. The official download link for this package is `http://downloads.mariadb.org/interstitial/mariadb-10.3.14/source/mariadb-10.3.14.tar.gz` which automatically derives to one of the mirrors. `wget` is able to hop to the designated mirror but EB can't. So I used the same mirror found in previous easyconfigs of MariaDB.
The releases in Github lack features compared to the releases in mariadb.org, for instance the RocksDB engine is not there.

File: easybuild/easyconfigs/m/MariaDB/MariaDB-10.3.14-foss-2019a.eb (hunk 3 above)
Comment (lexming) [reply easybuild/easyconfigs/m/MariaDB/MariaDB-10.3.14-foss-2019a.eb]: Both MariaDB Connector/C and RocksDB engine are included in the tarball.

**Comment by migueldiascosta** [comment]
Test report by @migueldiascosta
**SUCCESS**
Build succeeded for 6 out of 6 (6 easyconfigs in this PR)
sms - Linux centos linux 7.6.1810, AMD EPYC 7601 32-Core Processor, Python 2.7.5
See https://gist.github.com/6de24f9fddcfa039fe901347829b6f7d for a full test report.

**Comment by migueldiascosta** [comment]
Going in, thanks @lexming!

**Pull request merged**